packages/ui/src/modal-side/ModalSide.tsx

```diff
@@ -0,0 +1,119 @@
+"use client";
+
+import { useEffect, useRef, ReactNode, useState } from "react";
+import {
+  Dialog,
+  DialogTrigger,
+  Heading,
+  Modal,
+  ModalOverlay,
+} from "react-aria-components";
+
+import { TESTID_MODAL_SIDE } from "@repo/constants/src/testids";
+
+import "./ModalSide.css";
+import Icon from "../icon/Icon";
+import Button from "../button/Button";
+import { parseStringToComponents } from "web/utils/string";
+
+export interface ModalSideProps {
+  triggerContent: ReactNode;
+  sections: { number: string; header: string; content: string }[];
+  scrollToSection?: string;
+  "data-testid"?: string;
+}
+
+export default function ModalSide({
+  triggerContent,
+  sections,
+  scrollToSection,
+  "data-testid": testid = TESTID_MODAL_SIDE,
+  ...props
+}: ModalSideProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [focusSection, setFocusSection] = useState(scrollToSection);
+  const [highlightedSection, setHighlightedSection] = useState<string | null>(
+    null
+  );
+
+  const modalRef = useRef<HTMLDivElement>(null);
+  const sectionRefs = useRef<{ [key: string]: HTMLElement | null }>({});
+
+  useEffect(() => {
+    if (focusSection && sectionRefs.current[focusSection]) {
+      const sectionElement = sectionRefs.current[focusSection];
+      sectionElement?.scrollIntoView({
+        behavior: "instant",
+        block: "center",
+      });
+      setHighlightedSection(focusSection);
+    }
+  }, [isOpen, focusSection]);
+
+  return (
+    <DialogTrigger onOpenChange={setIsOpen}>
+      {triggerContent}
+
+      <ModalOverlay className="ui-ModalSide--Overlay">
+        <Modal
+          className="ui-ModalSide--Modal"
+          ref={modalRef}
+          data-testid={testid}
+        >
+          <Dialog className="ui-ModalSide--AriaDialog">
+            {({ close }) => (
+              <>
+                <header className="ui-ModalSide--Header">
+                  <Button
+                    isIconButton
+                    variant="tertiary"
+                    className="ui-ModalSide--CloseButton"
+                    onPress={close}
+                    data-testid={`${testid}-close-button`}
+                  >
+                    <Icon type="close" />
+                  </Button>
+                </header>
+                <main className="ui-ModalSide--Content">
```

Comment:
I like the semantic updates for a lot of this, but there is already a main on the page and there can only be one main element on a page semantically unless the other one is hidden. Ours is still in the DOM below the modal.